lib/grpc-lwt/client.ml

```diff
@@ -48,9 +37,24 @@ let call ~service ~rpc ?(scheme = "https") ~handler ~do_request
     match code with
     | None -> ()
     | Some code ->
-        let message = H2.Headers.get headers "grpc-message" in
-        let status = Grpc.Status.v ?message code in
-        Lwt.wakeup_later status_notify status
+        match Lwt.state status with 
+        | Sleep -> 
+          let message = H2.Headers.get headers "grpc-message" in
+          let status = Grpc.Status.v ?message code in
+          Lwt.wakeup_later status_notify status
+        | _ -> (* This should never happen, but just in case. *) ()
+  in
+  let response_handler (response : H2.Response.t) body =
+    Lwt.wakeup_later read_body_notify body;
+    Lwt.async (fun () ->
+        if response.status <> `OK then (
+          Lwt.wakeup_later out_notify
+            (Error (Grpc.Status.v Grpc.Status.Unknown));
+          Lwt.return_unit)
+        else
+          let+ handler_res in
+          Lwt.wakeup_later out_notify (Ok handler_res));
+    Lwt.async(fun () -> Lwt.return (trailers_handler response.headers))
```

Comment:
`Lwt.async` seems unnecessary here:
```suggestion
    trailers_handler response.headers
```

Reply:
I think you're right, let me double-check just in case.